### DISCUSSION TITLE
Home assistant not showing?

### DISCUSSION THREAD

**Discussion by Rabbit1543** [revk/xESP32-Faikout#127] [General]
Question on what to expect in home assistant if the device is not yet hooked up to an Daikin unit. Via MQTT Explorer I can see the topics for the 2 faikins esps I'm testing in to root, but they do not show anything under the home assistant topic.

**Comment by revk** [comment]
My sure as I have not tried that, but may be very limited.

**Comment by wurmtal868** [comment]
My remaining modules arrived yesterday and I tried configuring one of them with lab-bench supply.
I was able to set up WiFi and MQTT and after a reboot the module showed "MQTT connected" and "System is offline".
Nothing is visible in Home Assistant. I can only see one topic being sent through MQTT:
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/7a13738b-f11e-4866-a13a-d1520e419cc5)

Now comes the strange part. Even when I plug this module into my indoor unit, I get this:
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/7fd12108-db44-4ac1-9944-b58b54b3a9a9)

And when I try to update the FW I get an OTA error:
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/162a167e-c346-44ea-ad04-1d06f4a67663)

I'm not sure if this related to the set up, but it is very strange. 
MQTT is connected but the module can't connect to the update server?

4 other modules, which I set up connected to an indoor unit, are working fine. Only the module below shows "System is offline" and can't update via OTA?
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/731538ed-8fdd-4daf-90b1-013f6eface8b)

**Reply by revk** [reply]
System off line could simply be a faulty module I guess, though I do check them before they are shipped.

The failure to update is rather odd. Can you confirm date/time when it tried and I'll check the web server logs. Ideally what IP you would be coming from as well, if you can.

**Comment by wurmtal868** [comment]
How can I get to the web server logs?

**Reply by revk** [reply]
I mean *I* would check *my* web server logs for your Faikin trying to access the update, and see if anything looks odd.

**Comment by wurmtal868** [comment]
My IP was 92.192.130.223 when sending the OTA request.

**Reply by revk** [reply]
Err, OK, I have one request from that, today, 10:50, trying to fetch `Faikin-8266.bin` which is not one of my builds. These are ESP32 boards! I don't have any other log entries for that IP at all.

**Reply by wurmtal868** [reply]
There date and time was today maybe 20-30 min before I wrote my comment. The must several successful request from the same IP yesterday when I updated the other modules.

**Reply by wurmtal868** [reply]
Is it possible that the module identifies as ESP8266?

**Reply by revk** [reply]
None of the models from amazon would claim to be 8266, unless you had done something to change their settings via MQTT.

I see the logs of updates on 27th, 1st, 2nd, and 3rd, from that IP.

You can send `setting/nameofdevice` via MQTT and it should reply with its settings.

**Reply by wurmtal868** [reply]
The only thing I did to the module was configuring it without being connected to the indoor unit. 
What seems to be strange, was the module did not show the new IP address after configuring WiFi. I had to look into my router to get the IP address. There I noticed that router tracked a number which was different form the number shown in the web GUI.

**Reply by revk** [reply]
OK, that does seem rather odd. It should not really be possible for the web UI to show a different IP address. Might be worth trying the `settings` thing to check what it has.

**Reply by wurmtal868** [reply]
I currently struggling with the "settings" message, I'm newbie with MQTT :-(

**Reply by wurmtal868** [reply]
I guess I would be much faster if hook up USB connector and check the serial console.

**Reply by revk** [reply]
Well, assuming your device has a name of some sort, you just sent an MQTT message with the topic `settings/name` where `name` is the name you gave the device. You can set the name in its web interface settings.

You should get a message back `setting/name` with a JSON payload.

**Reply by revk** [reply]
That may provide some clues, yes.

**Reply by wurmtal868** [reply]
I got the USB connector:
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/5fee31a1-65b1-4d4f-9914-4efde396f30f)
What's next?

**Reply by revk** [reply]
`idf.py monitor`

**Reply by wurmtal868** [reply]
Is this part of the repository?

**Reply by revk** [reply]
That is part of the ESP IDF. Surely just confirming the settings via MQTT would be simpler.

**Reply by wurmtal868** [reply]
How about using minicom?

**Reply by revk** [reply]
No idea, might work. Never tried to use that instead of the ESP IDF.

**Reply by wurmtal868** [reply]
Sorry, a friend came by for coffee and I could not continue with the investigation.

After plugging the module into my Linux machine I can see dev/ttyACM0 and could confirm that the module is using an ESP-S3
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/36f2c750-3fae-451e-a31e-aa9c7efc77b7)

minicom is showing some watchdog errors:
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/5af5207a-9fa4-476c-abd2-c89f510842ec)

However,
the module is no longer connecting to my WiFi :-(

**Reply by revk** [reply]
The task not found is not an issue. Well done getting serial logging.

Maybe try the upgrade again and I'll check the log again to confirm if it is the same unit asking for the 8266 file.

**Reply by wurmtal868** [reply]
The unit is no longer visible in my WiFi. 
The LED is constantly blinking.

I have installed the ESP-IDF in VS-Code but I'm not used to it. Tried to open the ESP folder of your repository which I downloaded a week ago and I got this:
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/0de73d36-e279-40af-af5b-3fc69a973151)

**Reply by revk** [reply]
I have no idea about vs-code. I have never seen that screen. I have installed on linux and mac in the past, following the instructions, command line, so I can't help much. I really am unsure what you might have done to it.

**Reply by wurmtal868** [reply]
I just followed the instructions from https://docs.espressif.com/projects/esp-idf/en/stable/esp32s3/get-started/index.html which recommend installing the IDF via VS-Code. I'm already using VS-Code for PlatformIO that's why I went that way.

**Reply by revk** [reply]
OK, no idea, is that windows? I have command line ESP IDF here, and with that it is possible to clone the Faikin GitHub and `make install` to reflash your own build of the code if needed.

**Reply by wurmtal868** [reply]
I'm on lubuntu-20.04,  wait a second, 
I'm 50% into the download.

**Reply by wurmtal868** [reply]
VS-Code is installing 
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/95f002c2-b660-4be3-acf1-93a4b7b532dc)

**Reply by revk** [reply]
But given no serious error from serial, it either should be on wifi and so available on a web config, or not, so making an access point. It should be possible to get it on MQTT from either.

**Reply by wurmtal868** [reply]
I don't see an AP nor is the module connecting to my WiFi. I guess something is completely messes up in the ROM. Probably the package was x-rayed while shipping form amazon-uk to Germany.

**Reply by wurmtal868** [reply]
When I run idf.py monitor I get an an error:
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/ccbb4764-5be8-4c91-8524-e4089ac0b2b3)

**Reply by revk** [reply]
You did a `--recursive` clone, yes?

If not, try `make update`

**Comment by wurmtal868** [comment]
I did just git clone htpps: ... 
trying again with recursive....

**Reply by wurmtal868** [reply]
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/ab4942b4-3442-4b93-9773-6fda9b439708)

got an error when using --recursive

**Reply by wurmtal868** [reply]
The ESP32-RevK folder is empty :-(

**Reply by revk** [reply]
The `README.md` says `--recursive`, honest.

If you build yourself you will either need to (follow the instructions) to make a secure key for code signing or disable code signing (`make menuconfig`). If you disable code signing, then it should be able to auto update from my code.

You will want to `make s3` for the ESP32-S3 you have. And then `make install`

If you think something has been messed up, then `idf.py erase_flash` may be sensible before installing, to clear all settings, etc.

Hope that helps - I'll be out soon, so not replying for a bit I expect.

**Reply by revk** [reply]
Wow, no idea why it would fail to recurse. I don't suppose it explained any more? As I said, the other option, with the non recursive clone, was a simple `make update` which should recursively get everything.

<img width="1082" alt="Screenshot 2023-11-05 at 16 39 57" src="https://github.com/revk/ESP32-Faikin/assets/996983/f8b63141-e6a2-4548-868a-25c51cdcd256">

**Reply by wurmtal868** [reply]
I don't want to build it myself.  I just want the module to work. 
Can't I simply flash binaries from the release directory?

**Reply by revk** [reply]
They are in the release directory. But the `esptool` command is quite complex.

OK, this is what it does to flash it all. There is a binary for the S3 and boot loader in `ESP32/release` directory which you'd need to use instead of the bootloader.bin and Faikin.bin in this comment.

`esptool.py -p (PORT) -b 460800 --before default_reset --after hard_reset --chip esp32s3  write_flash --flash_mode dio --flash_size keep --flash_freq 40m 0x0 build/bootloader/bootloader.bin 0x8000 build/partition_table/partition-table.bin 0xd000 build/ota_data_initial.bin 0x10000 build/Faikin.bin`

**Reply by wurmtal868** [reply]
Where do I get partition_table, ota_data_initial bin and build/Faikin.bin?

**Reply by revk** [reply]
I said the Faikin.bin will be the S3 one in the realise. The others should be there, or a make should create them.

**Reply by wurmtal868** [reply]
Tried this command 
esptool.py -p /dev/ttyACM0 -b 460800 --before default_reset --after hard_reset --chip esp32s3  write_flash --flash_mode dio --flash_size keep --flash_freq 40m 0x0 release/Faikin-S3-MINI-N4-R2-bootloader.bin 0x8000 build/partition_table/partition-table.bin 0xd000 build/ota_data_initial.bin 0x10000 release/Faikin-S3-MINI-N4-R2.bin 
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/4d6de7bb-4bce-44de-804c-40fbfb4f0094)

**Reply by revk** [reply]
Did that now work.

If not, try idf.py erase_flash

Then flash it again.

**Reply by wurmtal868** [reply]
It's working :-) :-)
I could not find because it remained the WiFi credentials and picked up an IP which I which I thought was assigned to different module. My router even named it after the other module.

Thanks for helping me with fixing the module

**Comment by revk** [comment]
Just to be clear, this normally "just works". I am really puzzled how anything from your IP would try to get Faikin-8266 as that is not something that should ever try to upgrade from my server, not even the 8266 build (which someone else has made). Not being on WiFi and not doing an AP is not normal either, but no obvious error from the serial log. Almost like it has been flashed with some wrong code somehow. Having to "recover" is not something normally needed.

If all else fails, I'll try and sort a full binary image and a simple esptool command to flash from scratch. A recovery mode of sorts. That should be doable. But probably won't be until morning.

But may not reply much until then.

**Reply by wurmtal868** [reply]
Wait there is one thing that could cause the ESP8266 request. 
Before I got the first module from amazon-UK I was playing with the ESP8266 fork on a Wemos-D1mini. Maybe this module was trying to use your OTA server. I forgot about his module and it was still plugged into my system until I used the cable to debug the bad module.

**Comment by wurmtal868** [comment]
After fixing the module, I checked again if it gets detected in Home Assistant without being connected to an AC unit.
The result is same as before, Home Assistant will not detect the MQTT device until it was successfully connected to an indoor unit.

**Reply by revk** [reply]
Ok that is not surprising to be honest. The settings it offers HA all depend on what it "sees" as available via the S21 link to the aircon.

So, all good now?

**Reply by wurmtal868** [reply]
I'm all good now. 
Thanks a lot.

BTW: You just triggered another 50€ invest in BLE sensors, as I saw that you added locking of heat/cold modes :-)
You even showing it as automatic mode in Home Assistant and BLE temperature 
![image](https://github.com/revk/ESP32-Faikin/assets/32965910/aee60a8f-f838-422e-92f8-0ba2bda262fa)

Thanks for the great support!

**Comment by wurmtal868** [comment]
FYI I powered up my D1-mini. If you see now see incoming ESP8266 OTA request you should contact the owner of the 8266-fork.

**Reply by wurmtal868** [reply]
The module is flashed with the bin file files from his release folder.

**Reply by wurmtal868** [reply]
Are you seeing incoming requests from my address?

**Reply by revk** [reply]
Not since [05/Nov/2023:12:45:04 +0000] "GET /Faikin-8266.bin HTTP/1.1" 404 434 "-" "ESP32 HTTP Client/1.0"

**Reply by wurmtal868** [reply]
I just pushed the button  on D1 mini please check again?

**Reply by revk** [reply]
```
/var/log/apache2/access.log:92.192.130.223 - - [05/Nov/2023:19:08:45 +0000] "GET /Faikin-8266.bin HTTP/1.1" 404 434 "-" "ESP32 HTTP Client/1.0"
/var/log/apache2/access.log:92.192.130.223 - - [05/Nov/2023:19:08:47 +0000] "GET /Faikin-8266.bin HTTP/1.1" 404 434 "-" "ESP32 HTTP Client/1.0"
/var/log/apache2/access.log:92.192.130.223 - - [05/Nov/2023:19:08:48 +0000] "GET /Faikin-8266.bin HTTP/1.1" 404 434 "-" "ESP32 HTTP Client/1.0"
/var/log/apache2/access.log:92.192.130.223 - - [05/Nov/2023:19:08:49 +0000] "GET /Faikin-8266.bin HTTP/1.1" 404 434 "-" "ESP32 HTTP Client/1.0"
/var/log/apache2/access.log:92.192.130.223 - - [05/Nov/2023:19:08:51 +0000] "GET /Faikin-8266.bin HTTP/1.1" 404 434 "-" "ESP32 HTTP Client/1.0"
/var/log/apache2/access.log:92.192.130.223 - - [05/Nov/2023:19:08:53 +0000] "GET /Faikin-8266.bin HTTP/1.1" 404 434 "-" "ESP32 HTTP Client/1.0"
/var/log/apache2/access.log:92.192.130.223 - - [05/Nov/2023:19:08:54 +0000] "GET /Faikin-8266.bin HTTP/1.1" 404 434 "-" "ESP32 HTTP Client/1.0"
/var/log/apache2/access.log:92.192.130.223 - - [05/Nov/2023:19:09:02 +0000] "GET /Faikin-8266.bin HTTP/1.1" 404 434 "-" "ESP32 HTTP Client/1.0"
/var/log/apache2/access.log:92.192.130.223 - - [05/Nov/2023:19:09:03 +0000] "GET /Faikin-8266.bin HTTP/1.1" 404 434 "-" "ESP32 HTTP Client/1.0"
```

**Reply by revk** [reply]
Yeh, may have to have a word over their defaults!

**Reply by revk** [reply]
And well done, sorting things!

**Reply by wurmtal868** [reply]
That was me, I will retire the module now, as it caused a lot of confusion when debugging the other module.

**Reply by wurmtal868** [reply]
Again thanks a lot for helping me fixing  my module!
I will recommend your modules to my friends. 
There are at least 10 more units that are waiting to be freed from the Daikin Cloud :-)

Having the auto mode with remote sensing is a lot better than anything Daikin has to offer with their cloud!

**Reply by revk** [reply]
Stock on Amazon is OK, but planning on sending a load more next week.
